lib/plug/router/utils.ex

```diff
@@ -88,10 +88,15 @@ defmodule Plug.Router.Utils do
   """
   def forward(%Plug.Conn{path_info: path, script_name: script} = conn, new_path, target, opts) do
     {base, split_path} = Enum.split(path, length(path) - length(new_path))
-    conn = target.call(%{conn | path_info: split_path, script_name: script ++ base}, opts)
-    %{conn | path_info: path, script_name: script}
```

Comment:
Let's please keep this line as before as the map update syntax is much more efficient.